pkg/component/controller/controllersleasecounter.go

```diff
@@ -69,18 +67,19 @@ func (l *K0sControllersLeaseCounter) Start(ctx context.Context) error {
 	}
 	leaseName := fmt.Sprintf("k0s-ctrl-%s", nodeName)
 
-	leasePool, err := leaderelection.NewLeasePool(ctx, client, leaseName, l.InvocationID,
-		leaderelection.WithLogger(log),
-		leaderelection.WithContext(ctx))
+	leasePool, err := leaderelection.NewLeasePool(client, leaseName, l.InvocationID,
+		leaderelection.WithLogger(log))
 	if err != nil {
 		return err
 	}
-	events, cancel, err := leasePool.Watch()
+
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
I don't understand this change.
Why shouldn't the `context.WithCancel` be based on the one `Start`gets as an argument? If that context is Done I think this should be done too.

Reply:
This is one of the big issues that exists with the component model in k0s. See #4357 and its predecessor #1844.